src/Fieldset.php

```diff
@@ -382,11 +382,15 @@ public function prepareElement(FormInterface $form)
      */
     public function populateValues($data)
```

Comment:
Does it really accept Traversable? Looks like docblock is wrong here

Reply:
As mentioned in my comments, it was not implemented correctly and tests were using just ArrayObject (so ArrayAccess), but for any custom Traversable it was not working. I've updated tests and align the code to work with Traversable.

We cannot remove Traversable  here as it would be BC break, because before we accept it with a bug, but accept it. Check other tests and condition of I've updated below.